batch/batch/worker/worker.py

```diff
@@ -1430,18 +1422,35 @@ async def run(self):
                         for secret in self.secrets:
                             populate_secret_host_path(self.secret_host_path(secret), secret['data'])
 
-                with self.step('adding gcsfuse bucket'):
-                    if self.gcsfuse:
-                        populate_secret_host_path(self.credentials_host_dirname(), self.credentials.secret_data)
-                        for b in self.gcsfuse:
-                            bucket = b['bucket']
-                            await add_gcsfuse_bucket(
-                                mount_path=self.gcsfuse_path(bucket),
-                                bucket=bucket,
-                                key_file=self.credentials_host_file_path(),
-                                read_only=b['read_only'],
+                with self.step('adding cloudfuse support'):
+                    if self.cloudfuse:
+                        os.makedirs(self.cloudfuse_base_path())
+
+                        await check_shell_output(
+                            f'xfs_quota -x -c "project -s -p {self.cloudfuse_base_path()} {self.project_id}" /host/'
+                        )
```

Comment:
This seems not right to me. Wouldn't this effectively include the entire bucket's contents in the user's xfs quota? I'm not sure I even understand how that would work. It does seem important to put the temporary directory under a quota restriction.

Reply:
I'm not 100% sure, but my understanding was that the contents on disk were just cached files. I don't think it's the entire bucket. Nonetheless, maybe the fuse file system returns the full size of a directory which would be a problem if that's what xfs uses to track storage space. My guess is they are completely independent.

Reply:
Just to be clear. The temp directory is for Azure only. It's an unused directory for GCP